src/Widgets/WindowClone.vala

```diff
@@ -290,68 +286,40 @@ public class Gala.WindowClone : Clutter.Actor {
         in_slot_animation = true;
         place_widgets (outer_rect.width, outer_rect.height, initial_scale);
 
+        new GesturePropertyTransition (this, gesture_tracker, "x", null, (float) target_x).start (with_gesture);
+        new GesturePropertyTransition (this, gesture_tracker, "y", null, (float) target_y).start (with_gesture);
+        new GesturePropertyTransition (this, gesture_tracker, "width", null, (float) outer_rect.width).start (with_gesture);
+        new GesturePropertyTransition (this, gesture_tracker, "height", null, (float) outer_rect.height).start (with_gesture);
+        new GesturePropertyTransition (this, gesture_tracker, "shadow-opacity", (uint8) 255, (uint8) 0).start (with_gesture);
+        new GesturePropertyTransition (window_icon, gesture_tracker, "opacity", 255u, 0u).start (with_gesture, () => {
+            in_slot_animation = false;
+            place_widgets (outer_rect.width, outer_rect.height, target_scale);
+        });
+
         GestureTracker.OnUpdate on_animation_update = (percentage) => {
-            var x = GestureTracker.animation_value (initial_x, target_x, percentage);
-            var y = GestureTracker.animation_value (initial_y, target_y, percentage);
             var width = GestureTracker.animation_value (initial_width, outer_rect.width, percentage);
             var height = GestureTracker.animation_value (initial_height, outer_rect.height, percentage);
             var scale = GestureTracker.animation_value (initial_scale, target_scale, percentage);
-            var opacity = GestureTracker.animation_value (255f, 0f, percentage);
-
-            set_size (width, height);
-            set_position (x, y);
 
-            window_icon.opacity = (uint) opacity;
             set_window_icon_position (width, height, scale, false);
             place_widgets ((int)width, (int)height, scale);
-
-            shadow_opacity = (uint8) opacity;
         };
```

Comment:
Is it possible to move this to one of the done callbacks?

Reply:
Unfortunately not since it has to update during the animation